### DISCUSSION TITLE
Dispatch inside handleSubmit doesn't work

### DISCUSSION THREAD

**Discussion by caio-borghi-yapi** [react-hook-form/react-hook-form#8858] [Q&A]
I'm trying to update the value of a variable when the user submits a form but it's not working, the update does not affect the context of onValid function.

**Comment by Moshyfawn** [comment]
You'll have to provide more context and repo than that ;)

**Reply by caio-borghi-yapi** [reply]
Created one that explains what's happening

https://codesandbox.io/s/onvalid-always-old-state-4or4ie?file=/src/App.js

**Comment by caio-borghi-yapi** [comment]
On my real project, I'm updating a redux flag inside `customSubmit` function but when the `onValid` is executed, the value is always old.
